Add blt_export_tpl_targets macro (#556)
LLNL/blt: docs/tutorial/exporting_targets.rst

    # Add the example_1 target to the example-targets export set
    install(TARGETS example_1 EXPORT example-targets)

    # Add BLT's targets to the same export set - this is required
    # because the mpi target was marked exportable
    blt_export_tpl_targets(EXPORT example-targets)

> To avoid collisions with projects that import "example-targets", we can provide

Reviewer: ```suggestion
To avoid collisions with projects that import "example-targets", it is recommended to provide
```
